modules/vpc-sc/schemas/ingress-policy.schema.json

```
@@ -0,0 +1,80 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "VPC-SC ingress policy",
+  "type": "object",
+  "required": [
+    "from",
+    "to"
+  ],
+  "properties": {
+    "from": {
+      "type": "object",
```

```suggestion
      "type": "object",
      "additionalProperties": false,
      "required": [
        "access_levels"
      ],
```
Additional properties to catch if someone misspells the attribute.

Required - I vaguely remember that not providing access level bite me some time ago with not working policy. Encoding that in schema will be of a great help.

---

for ingress policies, access levels are not required: if you specify resources (i.e. projects) then you don't need access levels; if you don't have resources, then you need either a star of a list of access levels

TL;DR variable type is correct

---

I'm not saying it's not correct, I'm just saying it could be better, but IIUC, this means going into `anyOf`, where one requires `access_levels`  and the other that requires `resources`, so then we encode that one of them is provided.

But it will be PITA to encode those rules in JSON Schema, goes outside what we can express in HCL, so I would just keep in mind of this limitation.